### PR TITLE
Create RELEASING.md

### PR DESCRIPTION
**Pull request by moolitayer**

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Releasing Kubeclient
+
+## Versioning
+Kubeclient release versioning follows [semvar](https://semver.org/).
```
Comment (cben) [inline RELEASING.md]: s/semvar/SemVer/

File: RELEASING.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+At some point in time it is decided to release version x.y.z.
+Prerequisite: A [changelog](CHANGELOG.md) is generated & commited to the release branch.
+
+```bash
+# USERNAME=bob
```
Comment (cben) [inline RELEASING.md]: why commented out? (you use it below)
maybe call it RUBYGEMS_USERNAME for clarity.

File: RELEASING.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+#
+bundle install
+bundle exec rake test rubocop
+
+#
+# Grabbing an authentication token for rubygems.org api
+#
+curl -u $USERNAME https://rubygems.org/api/v1/api_key.yaml > ~/.gem/credentials; chmod 0600 ~/.gem/credentials
+
+cat /home//.gem/credentials
```
Comment (cben) [inline RELEASING.md]: ~/.gem/credentials

File: RELEASING.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+---
+:rubygems_api_key: ****
+
+#
+# Bump lib/kubeclient/version.rb and push the commit
+#
+gem bump --version $RELEASE_VERSION --push --destination $GIT_REMOTE
+# Merge the version bump pull request
```
Comment (cben) [inline RELEASING.md]: does above command create a PR?  I think it just pushes commit to master / $RELEASE_BRANCH?

Comment (moolitayer) [reply RELEASING.md]: Depends on the remote you are using

File: RELEASING.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+#
+# Release the gem
+#
+gem release
+
+#
+# Or in one command
+#
+gem bump --version 0.0.4 --destination $GIT_REMOTE --tag --release
```
Comment (cben) [inline RELEASING.md]: $RELEASE_VERSION

File: RELEASING.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+# Install the release gem
+#
+gem install gem-release
+
+
+#
+# Checking out the release branch
+#
+git fetch $GIT_UPSTREAM
+git checkout $GIT_UPSTREAM/$RELEASE_BRANCH -b $RELEASE_BRANCH
```
Comment (cben) [inline RELEASING.md]: should also do `git status` / `git reset --hard` to ensure no uncommitted changes.

File: RELEASING.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+curl -u $RUBYGEMS_USERNAME https://rubygems.org/api/v1/api_key.yaml > ~/.gem/credentials; chmod 0600 ~/.gem/credentials
+
+cat ~/.gem/credentials
+---
+:rubygems_api_key: ****
+
+#
+# Bumping the gem's version can be done manually, or by using
+#
+gem bump --version $RELEASE_VERSION --push --destination $GIT_REMOTE
```
Comment (cben) [inline RELEASING.md]: Checked experimentally (on another repo) how gem `--push` determines destination *branch*:
It executes `git push $GIT_REMOTE`, which requires currently checked out branch to track an "upstream" branch.  The above `git checkout` command does configure such tracking (can be verified by `git branch -vv`, displays upstream branch in square brackets).

So this works, but IMHO too fragile, I prefer doing:
```
gem bump --version $RELEASE_VERSION
git show
git push $GIT_REMOTE $RELEASE_BRANCH
gem tag --destination $GIT_UPSTREAM
gem release
```

File: RELEASING.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+# ... Create & Merge A version bump pull request
```
Comment (cben) [inline RELEASING.md]: if we're doing bump via PR, should `git pull $GIT_UPSTREAM` after merging to tag correctly.
that also rules out the all-in-one `gem bump --version $RELEASE_VERSION --destination $GIT_REMOTE --tag --release`.
(I'm ok with either PR or direct bump push, but let's decide & document just one flow...)

**Comment by moolitayer** [comment]
Hi @cben could you please take another look.

I'd like to release soon, it makes sense to me to merge this only after actually releasing.

File: RELEASING.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
+# Releasing Kubeclient
+
+## Versioning
+Kubeclient release versioning follows [SemVer](https://semver.org/).
+At some point in time it is decided to release version x.y.z.
+Prerequisite: A [changelog](CHANGELOG.md) is generated & merged into the release branch.
```
Comment (cben) [inline RELEASING.md]: I think we always want a CHANGELOG covering all releases on master (and if releasing a y-stream branch, include at least that branch releases in CHANGELOG on that branch too

File: RELEASING.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+:rubygems_api_key: ****
+
+#
+# Bumping the gem's version can be done manually, or by using
+#
+gem bump --version $RELEASE_VERSION
+git show # View version bump change.
+
+#
+# create & merge a version bump PR
```
Comment (cben) [inline RELEASING.md]: this is unclear.  what I think this meant is:
```
git push $GIT_REMOTE $RELEASE_BRANCH
# Now go to github, open PR from your fork $RELEASE_BRANCH to upstream $RELEASE_BRANCH.
# Review/Merge the PR on github.
git pull --ff-only $GIT_UPSTREAM
git show # make sure you received the PR merge commit.
```

**Comment by cben** [comment]
@moolitayer I think you had uncommitted edits to this, can you push them?

**Comment by cben** [comment]
@moolitayer ping
I want to do yet another release (to fix damage from my last one :-\ #338), can you push latest procedure?

**Pull request merged**